commands/sql/sql.drush.inc

```diff
@@ -529,7 +531,28 @@ function drush_sql_get_class($db_spec = NULL) {
   // Try a few times to quickly get $db_spec.
   if (!empty($db_spec)) {
     if (!empty($db_spec['driver'])) {
-      return drush_get_class(array('Drush\Sql\Sql', 'Drupal\Driver\Database\\' .  $db_spec['driver'] . '\Drush'), array($db_spec), array($db_spec['driver']));
+      $class_name = [
+        'Drush\Sql\Sql',
+        'Drupal\Driver\Database\\' . $db_spec['driver'] . '\Drush',
+      ];
+      try {
+        $class = drush_get_class($class_name, [$db_spec], [$db_spec['driver']]);
+        return $class;
+      }
+      catch (ArgumentCountError $e) {
+        /** @var \Drush\Log\Logger $logger */
+        $logger = drush_get_context('DRUSH_LOGGER');
```

Comment:
Fix this on `master` first. Get the logger from the `Drush` class, as `drush_get_context` has been removed.

Reply:
Never mind about fixing this in master; I think this will never happen on master because master won't allow a site-local Drush to be used with a different Drupal site.